sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedWriteSuite.scala

```diff
@@ -48,16 +49,40 @@ abstract class BucketedWriteSuite extends QueryTest with SQLTestUtils {
     intercept[AnalysisException](df.write.bucketBy(2, "k").saveAsTable("tt"))
   }
 
-  test("numBuckets be greater than 0 but less than 100000") {
+  test("numBuckets be greater than 0 but less than default bucketing.maxBuckets (100000)") {
     val df = Seq(1 -> "a", 2 -> "b").toDF("i", "j")
 
-    Seq(-1, 0, 100000).foreach(numBuckets => {
-      val e = intercept[AnalysisException](df.write.bucketBy(numBuckets, "i").saveAsTable("tt"))
-      assert(
-        e.getMessage.contains("Number of buckets should be greater than 0 but less than 100000"))
+    Seq(-1, 0, 100001).foreach(numBuckets => {
```

Comment:
nit: Only two parts are necessary to be updated for ease of tracking updates. Other changes look unnecessary.
`100000` -> `100001`
`less than 100000` -> `less than`